libavcodec/jpeg2000dec.c

```diff
@@ -2178,22 +2451,43 @@ static int jpeg2000_read_main_headers(Jpeg2000DecoderContext *s)
             if (!s->tile)
                 s->numXtiles = s->numYtiles = 0;
             break;
+        case JPEG2000_CAP:
+            if (!s->ncomponents) {
+                av_log(s->avctx, AV_LOG_ERROR, "CAP marker segment shall come after SIZ\n");
```

Comment:
Definitely should not be an error, but a warning.

Reply:
Accepted.